apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/landing/route.ts

```diff
@@ -0,0 +1,52 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { signIn } from 'apps/payments/next/auth';
+import { redirect } from 'next/navigation';
+import { NextRequest } from 'next/server';
+
+export const dynamic = 'force-dynamic'; // defaults to auto
+
+/**
+ * This landing route will initiate the OAuth no prompt signin
+ * attempt with FxA.
+ *
+ * Only when an unexpected error occurs, will the user be redirected
+ * to a generic error page.
+ *
+ * On successful authentication, the customer will be redirected
+ * to the /new page in a "Signed In" state. (i.e. FxA uid added to cart)
+ *
+ * On failure the customer will be redirected to /error/auth/signin
+ * where the error will be handled correctly, and ideally redirect
+ * the customer to the /new page in a "Signed Out" state. (i.e. no
+ * FxA uid added to the cart)
+ *
+ * This needs to be a route handler, since the `signIn` server
+ * action needs to be executed from a route handler.
+ */
+export async function GET(request: NextRequest) {
+  // Replace 'landing' in the URL with 'new'
+  const redirectTo = request.nextUrl.href
+    .split('/')
+    .toSpliced(-1, 1, 'new')
```

Comment:
TIL: [`.toSpliced()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced) is a thing (circa Node 20 and Firefox 115).

Although, I am confused at what it does... 
```js
let href = "https://foo.bar/one/two/three/?foo=baz";
let redirectTo = href.split("/").toSpliced(-1, 1, "new").join("/");
console.log(redirectTo); // https://foo.bar/one/two/three/new
```
It seems to be replacing the last element of the array with the replacement value. I probably had a weak example in about:blank where it's replacing the query string instead of the `/three/` path though.
So probably not a concern if we don't expect query params, but also might be unpredictable if the `request.nextUrl.href` has a trailing `/`:

```js
href = "https://foo.bar/one/two/three/";
let redirectTo = href.split("/").toSpliced(-1, 1, "new").join("/");
console.log(redirectTo); // https://foo.bar/one/two/three/new
```

Reply:
Thank you for the comments. I've updated the logic a bit, that should hopefully improve things.